build-advisories/src/Roave/SecurityAdvisories/Component.php

```diff
@@ -53,15 +59,44 @@ public function getName()
      */
     public function getConflictConstraint()
     {
-        // too simplified when multiple branches reference congruent versions
         return implode(
             '|',
             array_filter(array_map(
-                function (Advisory $advisory) {
-                    return $advisory->getConstraint();
+                function (VersionConstraint $versionConstraint) {
+                    return $versionConstraint->getConstraintString();
                 },
-                $this->advisories
+                $this->deDuplicateConstraints(array_merge(
+                    [],
+                    ...array_values(array_map(
+                        function (Advisory $advisory) {
+                            return $advisory->getVersionConstraints();
+                        },
+                        $this->advisories
+                    ))
+                ))
             ))
         );
     }
+
+    /**
+     * @param VersionConstraint[] $constraints
+     *
+     * @return VersionConstraint[]
+     */
+    private function deDuplicateConstraints(array $constraints)
+    {
+        restart:
+
+        foreach ($constraints as $constraint) {
+            foreach ($constraints as $key => $comparedConstraint) {
+                if (($constraint !== $comparedConstraint) && $constraint->contains($comparedConstraint)) {
```

Comment:
Yeah, but PHP operator precedence always confuses the heck out of me, so I preferred to keep them